security/pfSense-pkg-HoneyTrap/files/usr/local/pkg/honeytrap/honeytrap-plugin.inc

```diff
@@ -81,13 +81,25 @@ function honeytrap_install() {
 }
 
 function honeytrap_deinstall() {
+    update_status('Removing HoneyTrap')
 	if (is_service_running('honeytrap')) {
-		log_error('Stopping service honeytrap');
+		update_status('Stopping HoneyTrap service');
 		stop_service('honeytrap');
 	}
 
-	unlink_if_exists('/usr/local/etc/rc.d/honeytrap-daemon.sh');
-	rmdir_recursive('/usr/local/pkg/honeytrap/.data/');
-	rmdir_recursive("{$g['varlog_path']}/honeytrap/", true);
+    global $config;
+	if (!isset($config['installedpackages']['honeytrap'])) {
+		init_config_arr(array('installedpackages', 'honeytrap', 'config', 0));
```

Comment:
init_config_arr() can be always called, it will check if any piece of that array is missing and create, no need to check before call it